### ISSUE TITLE
Accuracy metric throwing out-of-range value (>1)

### ISSUE THREAD

**Issue by Vichoko** [PyTorchLightning/pytorch-lightning#4150]
## 🐛 Bug

Using class-based `metrics.Accuracy(num_classes)` with a fixed number of classes calculated in train and eval step outputs accuracy bigger than one.

`Epoch 11:   2%|▏         | 66/2775 [02:00<1:22:17,  1.82s/it, loss=2.367, v_num=1, train_loss=2.36, train_acc=1.67, val_loss=2.41, val_acc=1.27]`

### To Reproduce

The issue appears inconsistently.

### Expected behavior

Accuracy should be a value between 0 and 1.

### Environment

 - PyTorch Version (e.g., 1.0): 1.6.0
 - OS (e.g., Linux): Ubuntu
 - How you installed PyTorch (`conda`, `pip`, source): Conda
 - Python version: 3.8.5
- Pytorch-lightning version: 0.9.0
 - Any other relevant information: I'm using multi-gpu with ddp

### Additional context

I'm getting these warnings from the accuracy metric:

```
/miniconda3/envs/py385/lib/python3.8/site-packages/pytorch_lightning/utilities/distributed.py:37: UserWarning: You have set 32 number of classes if different from predicted (4) and target (29) number of classes
  warnings.warn(*args, **kwargs)
```

Actually I just wanted to report it even if I don't have a reproduce because I didn't see any other reports of this. 
Maybe accuracy should have a warning in these cases, but I have no clue where this out-of-bounds value is coming from in the first place.

Meanwhile, I'll switch to an ad-hoc accuracy calculation to check if the problem persists.

**Comment by teddykoker** [comment]
PyTorch Lightning 1.0 has a new accuracy metric that is correctly implemented and rigorously tested, I would recommend updating to that.

**Comment by Vichoko** [comment]
Okay, I'll update and will keep informed if the issue persists.
